### PR TITLE
adding 'Version' to HostStatus

### PR DESCRIPTION
**Pull request by brettsam**
Addresses #403. We'll also need to update AppVeyor to patch the build number into AssemblyFileVersion.

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Models/HostStatus.cs [hunk 1]
```diff
@@ -2,16 +2,41 @@
...
+using System.Reflection;
 using Newtonsoft.Json;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Models
 {
     public class HostStatus
     {
+        private static string _assemblyFileVersion;
+
+        public HostStatus()
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Models/HostStatus.cs]: We can move this logic to a static initializer and no longer have to worry about the null check

File: src/WebJobs.Script.WebHost/Models/HostStatus.cs [hunk 2]
```diff
@@ -2,16 +2,41 @@
...
 
 using System.Collections.ObjectModel;
+using System.Reflection;
 using Newtonsoft.Json;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Models
 {
     public class HostStatus
     {
+        private static string _assemblyFileVersion;
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Models/HostStatus.cs]: Make this `readonly` (also see comment below)

File: src/WebJobs.Script.WebHost/Models/HostStatus.cs [hunk 3]
```diff
@@ -2,16 +2,41 @@
...
+        /// <summary>
+        /// Gets the host version.
+        /// </summary>
+        [JsonProperty(PropertyName = "version", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string Version { get; private set; }
+
+        private static string GetExecutingAssemblyFileVersion()
+        {
+            AssemblyFileVersionAttribute fileVersionAttr = Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyFileVersionAttribute>();
+            return fileVersionAttr != null && fileVersionAttr.Version != null ? fileVersionAttr.Version : "Unknown";
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Models/HostStatus.cs]: nit: `fileVersionAttr?.Version ?? "Unknown"` ?

**Comment by fabiocav** [comment]
Added a couple of small comments

File: src/WebJobs.Script.WebHost/Models/HostStatus.cs (hunk 1 above)
Comment (brettsam) [inline src/WebJobs.Script.WebHost/Models/HostStatus.cs]: Made it a public static readonly field in my latest commit -- see if you like that any better :-)

File: src/WebJobs.Script.WebHost/Models/HostStatus.cs (hunk 3 above)
Comment (brettsam) [inline src/WebJobs.Script.WebHost/Models/HostStatus.cs]: Good idea. That was some old code from another repo I had copied.

**Comment by fabiocav** [comment]
Looks good!
One small comment is, which is a bit of a nit, but I think it would be a better approach to have the version field made private and expose a property instead, but it's good to :shipit: !

**Comment by brettsam** [comment]
I tried some other things:
- a public static property initialized in static ctor, but it wouldn't serialize
- a public instance property exposing a private static field. FxCop didn't like that, but I could always suppress it.

The field felt the simplest. If there's a better pattern, I can come back and change it.

**Pull request merged**